test/kraken.js

```diff
@@ -302,8 +300,45 @@ test('kraken', function (t) {
             t.ok(1, 'server stopped');
         });
 
+        // This listens on any random port the OS assigns.
+        // since we don't actually connect to it for this test, we don't care which.
+        //
+        // See https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback
+        // for more information
         server = app.listen(0);
         server.timeout = 0;
     });
 
+    t.test('shutdown headers', function (t) {
+        var app, server;
+
+        process.removeAllListeners('SIGTERM');
+
+        app = express();
+        app.use(kraken({ basedir: __dirname }));
+
+        app.on('start', function () {
+
+            app.removeAllListeners('shutdown');
+
+            app.once('shutdown', function () {
+                request(app).get('/').end(function (error, response) {
+                    t.ok(!error, 'no error.');
+                    t.equals(response.statusCode, 503, 'correct status code.');
```

Comment:
This won't work on 0.12 because the [`end` doesn't fire with any parameters](https://github.com/joyent/node/blob/4d9c81b7e2522c5d5d9d35058cbb0bce1228d360/lib/_http_outgoing.js#L522-L527).